### PR TITLE
Only print modified options when switching ap

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose
Previously, the modified options were printed on every `__call__`, which is excessive. Now, they are printed only when switching aero problems, since it's possible for a solver to use different options based on aero problems.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Testing
This works locally.

## Checklist
_Put an `x` in the boxes that apply._

- [ ] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Review by sseraj — APPROVED** [review]

File: adflow/pyADflow.py [hunk 1]
```diff
@@ -2546,10 +2542,12 @@ def setAeroProblem(self, aeroProblem, releaseAdjointMemory=True):
         # Tell the user if we are switching aeroProblems
         if self.curAP != aeroProblem:
             newAP = True
-            if self.comm.rank == 0:
```
Comment (anilyil) [inline adflow/pyADflow.py:2549]: why did you remove the comm.rank check here? I am guessing the current version will print this on all procs in comm now?

**Review by anilyil — CHANGES_REQUESTED** [review]
can you check and make sure the "switching to aero problem" print is not done on all procs. we just want the root doing it.

File: adflow/pyADflow.py (hunk 1 above)
Comment (marcomangano) [reply adflow/pyADflow.py:2549]: With the `baseclasses` refactoring, a [`pp` function that use pretty print on the sole root proc has been added](https://github.com/mdolab/baseclasses/blob/7ed195c5064b489325f74e4c6b551123b3b67a61/baseclasses/BaseSolver.py#L227-L238).
It is true however that the [pprint import](https://github.com/mdolab/adflow/blob/a310c0f8484b88dd74162de44f8f298d9be87b5b/adflow/pyADflow.py#L35) in `pyADflow.py` is ambiguous at this point, and we might want to remove it altogether and use the provided `baseclasses` method.
@anilyil @nwu63 @sseraj does it make sense?

Comment (anilyil) [reply adflow/pyADflow.py:2549]: oh thats cool. yeah the import in pyadflow is misleading

**Comment by nwu63** [comment]
I will fix the `pprint` import. Is the rest of the PR okay? I didn't change the rest of root print to use `self.pp` so maybe for consistency we want to revert back to the old code?

**Comment by marcomangano** [comment]
> I didn't change the rest of root print to use `self.pp` so maybe for consistency we want to revert back to the old code?

We can update them when we reformat the python scripts imho

**Review by anilyil — APPROVED** [review]

**Review by sseraj — APPROVED** [review]
I agree with @marcomangano. I think this is fine as is.
